dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java

```diff
@@ -170,5 +173,116 @@ public void Test_That_File_Asset_Gets_Stored_in_Cache_and_is_Not_Rebuilt_Everyti
       assertTrue("FileAssets should be the same Object", asset3 == asset4);
       
     }
+    
+    /**
+     * This tests that file assets saved to a random folder, e.g. /dasfasd/ will be returned from the database
+     * @throws Exception
+     */
+    @Test
+    public void test_that_file_asset_from_db_works()
+            throws Exception {
+
+        final User user = APILocator.systemUser();
+        final Folder parentFolder = new FolderDataGen().nextPersisted();
+    
+        List<String> fileNames = new ArrayList<>();
+        final int fileAssetSize=6;
+        
+        for(int i=0;i<fileAssetSize;i++) {
+            final java.io.File file = java.io.File.createTempFile("blah" + i, ".txt");
+            fileNames.add(file.getName());
+            FileUtil.write(file, "helloworld");
+            final FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(parentFolder, file);
+            fileAssetDataGen.setPolicy(IndexPolicy.DEFER);
+            fileAssetDataGen.nextPersisted();
+        }
+        
+        FileAssetSearcher searcher = FileAssetSearcher.builder().user(user).folder(parentFolder).respectFrontendRoles(false).build();
+                        
+
+        List<FileAsset> assets = APILocator.getFileAssetAPI().findFileAssetsByDB(searcher);
+        assert(assets.size()==fileAssetSize);
+        assets.forEach(a-> {
+            assert(fileNames.contains(a.getFileName()));
+        });
+
+    }
+    
+    /**
+     * This tests that file assets saved to a folder are not readable unless the user has read permissions to the parent folder itself
+     * @throws Exception
+     */
+    @Test
+    public void test_that_file_asset_from_db_respects_folder_permissions()
+            throws Exception {
+
+        final User user = APILocator.getUserAPI().getAnonymousUser();
+        final Folder parentFolder = new FolderDataGen().nextPersisted();
+    
+        List<String> fileNames = new ArrayList<>();
+        final int fileAssetSize=2;
+        
+        for(int i=0;i<fileAssetSize;i++) {
+            final java.io.File file = java.io.File.createTempFile("blah" + i, ".txt");
+            fileNames.add(file.getName());
+            FileUtil.write(file, "helloworld");
+            final FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(parentFolder, file);
+            fileAssetDataGen.setPolicy(IndexPolicy.DEFER);
+            fileAssetDataGen.nextPersisted();
+        }
+        
+        FileAssetSearcher searcher = FileAssetSearcher.builder().user(user).folder(parentFolder).respectFrontendRoles(false).build();
+                        
+
+        try {
+            APILocator.getFileAssetAPI().findFileAssetsByDB(searcher);
+        }
+        catch(DotRuntimeException e) {
+            assertTrue("We should have thrown a DotRuntimeException", e!=null);
```

Comment:
Codacy found an issue: [Use assertNull(x) instead of assertTrue(x==null), or assertNotNull(x) vs assertFalse(x==null)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6325390)